pubsub/kafka/kafka.go

```diff
@@ -62,6 +64,46 @@ func (p *PubSub) Subscribe(ctx context.Context, req pubsub.SubscribeRequest, han
 	return p.kafka.Subscribe(p.subscribeCtx)
 }
 
+func (p *PubSub) BulkSubscribe(ctx context.Context, req pubsub.SubscribeRequest, handler pubsub.BulkHandler) error {
+	p.kafka.AddTopicBulkHandler(req.Topic, adaptBulkHandler(handler))
+	maxBulkCount, err := strconv.Atoi(req.Metadata["maxBulkCount"])
+	if err != nil {
+		maxBulkCount = 20
+	}
+	maxBulkAwaitDurationMilliSeconds, err := strconv.Atoi(req.Metadata["maxBulkAwaitDurationMilliSeconds"])
+	if err != nil {
+		maxBulkAwaitDurationMilliSeconds = 20
+	}
+	maxBulkSizeBytes, err := strconv.Atoi(req.Metadata["maxBulkSizeBytes"])
+	if err != nil {
+		maxBulkSizeBytes = 20
+	}
+	p.kafka.AddBulkSubscribeConfig(maxBulkCount, maxBulkAwaitDurationMilliSeconds, maxBulkSizeBytes)
+
+	go func() {
+		// Wait for context cancelation
+		select {
+		case <-ctx.Done():
+		case <-p.subscribeCtx.Done():
+		}
+
+		// Remove the topic handler before restarting the subscriber
+		p.kafka.RemoveTopicBulkHandler(req.Topic)
+
+		// If the component's context has been canceled, do not re-subscribe
+		if p.subscribeCtx.Err() != nil {
+			return
+		}
+
+		err := p.kafka.BulkSubscribe(p.subscribeCtx)
+		if err != nil {
+			p.logger.Errorf("kafka pubsub: error re-subscribing: %v", err)
+		}
+	}()
+
+	return p.kafka.BulkSubscribe(p.subscribeCtx)
```

Comment:
What happens if the connection is reset? Will the component reconnect?

Reply:
Yes, if the connection with broker is stopped and then started back, component is reconnected.